test/PHPUnit/Controller/AbstractControllerTestCaseTest.php

```diff
@@ -248,6 +248,68 @@ public function testAssertNotMatchedRouteName()
         $this->assertNotMatchedRouteName('myroute');
     }
 
+    public function testAssertNoMatchedRoute()
+    {
+        $this->dispatch('/invalid');
+        $this->assertNoMatchedRoute();
+    }
+
+    public function testAssertNoMatchedRouteWithMatchedRoute()
+    {
+        $this->dispatch('/tests');
+        $this->setExpectedException('PHPUnit_Framework_ExpectationFailedException');
```

Comment:
Each of these should also provide a message to match (i.e., `'No route matched'`), in part to make it more clear what sort of failure is expected.

I can add those on merge, however.